src/entities/trades/curve/abi/abi.ts

```diff
@@ -58,6 +65,52 @@ export const CURVE_ETHXERC20_256_ABI: ContractInterface = [
   poolMethods['payable']['exchange_underlying(uint256,uint256,uint256,uint256)'],
 ]
 
+export const CURVE_METAUSD_ABI: ContractInterface = [
+  poolMethods['view']['fee'],
+  poolMethods['view']['get_dy_underlying(int128,int128,uint256)'],
+  poolMethods['view']['get_dy(int128,int128,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,uint256,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,int256,int256,address)'],
+  poolMethods['nonpayable']['exchange_underlying(int128,int128,uint256,uint256)'],
+  poolMethods['nonpayable']['exchange_underlying(int128,int128,uint256,uint256,address)'],
+]
+
+export const CURVE_PLAIN_ABI: ContractInterface = [
+  poolMethods['view']['fee'],
+  poolMethods['view']['get_dy(int128,int128,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,uint256,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,int256,int256,address)'],
+]
+
+export const CURVE_METABTC_ABI: ContractInterface = [
+  poolMethods['view']['fee'],
+  poolMethods['view']['get_dy(int128,int128,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,uint256,uint256)'],
+  poolMethods['nonpayable']['exchange(int128,int128,int256,int256,address)'],
+  poolMethods['nonpayable']['exchange_underlying(int128,int128,uint256,uint256)'],
+  poolMethods['nonpayable']['exchange_underlying(int128,int128,uint256,uint256,address)'],
+]
+
+export const CURVE_POOL_ABI_MAP: Record<string, ContractInterface> = {
+  ['metabtc']: CURVE_METABTC_ABI,
+  ['metabtcbalances']: CURVE_METABTC_ABI,
+  ['metausd']: CURVE_METAUSD_ABI,
+  ['v1metausd']: CURVE_METAUSD_ABI,
+  ['metausd-fraxusdc']: CURVE_METAUSD_ABI,
+  ['metausdbalances']: CURVE_METAUSD_ABI,
+  ['plain2balances']: CURVE_PLAIN_ABI,
+  ['plain2basic']: CURVE_PLAIN_ABI,
+  ['plain2eth']: CURVE_PLAIN_ABI,
+  ['plain2optimized']: CURVE_PLAIN_ABI,
+  ['plain3balances']: CURVE_PLAIN_ABI,
+  ['plain3basic']: CURVE_PLAIN_ABI,
+  ['plain3eth']: CURVE_PLAIN_ABI,
+  ['plain3optimized']: CURVE_PLAIN_ABI,
+  ['plain4balances']: CURVE_PLAIN_ABI,
+  ['plain4basic']: CURVE_PLAIN_ABI,
+  ['plain4eth']: CURVE_PLAIN_ABI,
+  ['plain4optimized']: CURVE_PLAIN_ABI,
+}
```

Comment:
How much effort does this require to maintain?

Reply:
I'm not sure but I would expect little effort.
To maintain this all that is needed is to get each pool's ABI and compare with what we have implemented.
Worst case scenario, pn the off-chance the curve team changes the ABI/curve types, I would estimate it to take 1 or 2 weeks to completely overhaul everything. Best case scenario, they change only 1 of the pools, it would take 1 or 2 days.
Keep in mind these are my personal estimations and estimations are hand-waved.